src/lib.rs

```diff
@@ -114,4 +343,165 @@ impl Config {
         self.key_path = Some(key_path.into());
         self
     }
+
+    pub fn timeout(mut self, timeout: Duration) -> Self {
+        self.timeout = timeout;
+        self
+    }
+}
+
+pub type KvFuture<Resp> = Box<Future<Item = Resp, Error = Error>>;
+
+/// A convenience trait for expressing ranges.
+///
+/// In TiKV, keys are an ordered sequence of bytes. This means we can have ranges over those
+/// bytes. Eg `001` is before `010`.
+///
+/// This trait has implementations for common range types like `a..b`, `a..=b` where `a` and `b`
+/// `impl Into<Key>`. You could implement this for your own types.
+///
+/// ```rust
+/// use tikv_client::{KeyRange, Key};
+/// use std::ops::{Range, RangeInclusive, RangeTo, RangeToInclusive, RangeFrom, RangeFull, Bound};
+///
+/// let explict_range: Range<Key> = Range { start: Key::from("Rust"), end: Key::from("TiKV") };
+/// let from_explict_range = explict_range.into_bounds();
+///
+/// let range: Range<&str> = "Rust".."TiKV";
+/// let from_range = range.into_bounds();
+/// assert_eq!(from_explict_range, from_range);
+///
+/// let range: RangeInclusive<&str> = "Rust"..="TiKV";
+/// let from_range = range.into_bounds();
+/// assert_eq!(
+///     (Bound::Included(Key::from("Rust")), Bound::Included(Key::from("TiKV"))),
+///     from_range
+/// );
+///
+/// let range_from: RangeFrom<&str> = "Rust"..;
+/// let from_range_from = range_from.into_bounds();
+/// assert_eq!(
+///     (Bound::Included(Key::from("Rust")), Bound::Unbounded),
+///     from_range_from,
+/// );
+///
+/// let range_to: RangeTo<&str> = .."TiKV";
+/// let from_range_to = range_to.into_bounds();
+/// assert_eq!(
+///     (Bound::Unbounded, Bound::Excluded(Key::from("TiKV"))),
+///     from_range_to,
+/// );
+///
+/// let range_to_inclusive: RangeToInclusive<&str> = ..="TiKV";
+/// let from_range_to_inclusive = range_to_inclusive.into_bounds();
+/// assert_eq!(
+///     (Bound::Unbounded, Bound::Included(Key::from("TiKV"))),
+///     from_range_to_inclusive,
+/// );
+///
+/// let range_full: RangeFull = ..;
+/// let from_range_full = range_full.into_bounds();
+/// assert_eq!(
+///     (Bound::Unbounded, Bound::Unbounded),
+///     from_range_full
+/// );
+/// ```
+///
+/// **But, you should not need to worry about all this:** Many functions accept a `impl KeyRange`
+/// which means all of the above types can be passed directly to those functions.
+pub trait KeyRange: Sized {
+    fn into_bounds(self) -> (Bound<Key>, Bound<Key>);
+    fn into_keys(self) -> Result<(Key, Option<Key>)> {
+        range_to_keys(self.into_bounds())
+    }
+}
+
+fn range_to_keys(range: (Bound<Key>, Bound<Key>)) -> Result<(Key, Option<Key>)> {
+    let start = match range.0 {
+        Bound::Included(v) => v,
+        Bound::Excluded(mut v) => {
+            let len = v.len();
+            if len > 0 {
+                v.deref_mut().get_mut(len - 1).map(|v| {
+                    *v += 1;
+                    v
+                });
```

Comment:
Maybe this is better? It can handle overflow and do not use `map`
```rust
match v.last_mut() {
    None | Some(&mut U8_MAX) => v.push(0),
    Some(v) => *v+=1,
}   
v   
```